homeassistant/components/nice_go/coordinator.py

```diff
@@ -220,4 +227,39 @@ async def on_data(self, data: dict[str, Any]) -> None:
     async def on_connected(self) -> None:
         """Handle the websocket connection."""
         _LOGGER.debug("Connected to the websocket")
+        self.connected = True
+
         await self.api.subscribe(self.organization_id)
+
+        if not self.last_update_success:
+            self.async_set_updated_data(self.data)
+
+    async def on_connection_lost(self, data: dict[str, Exception]) -> None:
+        """Handle the websocket connection loss. Don't need to do much since the library will automatically reconnect."""
+        _LOGGER.debug("Connection lost to the websocket")
+        self.connected = False
+
+        # Give some time for reconnection
+        await asyncio.sleep(5)
+        if self.connected:
+            _LOGGER.debug("Reconnected, not setting error")
+            return
+
+        # There's likely a problem with the connection, and not the server being flaky
+        self.async_set_update_error(data["exception"])
```

Comment:
Is the subscription to be retained here?

Reply:
Yes, event subscription from `api.listen` persists through re-connection.

`api.subscribe` lets the server know what we're looking for, and is called upon re-connection under `on_connected`